syncer/syncer.go

```diff
@@ -635,27 +641,10 @@ func (s *Syncer) Run() error {
 }
 
 // Connect forms an outbound connection to a peer.
-func (s *Syncer) Connect(addr string) (*Peer, error) {
+func (s *Syncer) Connect(ctx context.Context, addr string) (*Peer, error) {
 	if err := s.allowConnect(addr, false); err != nil {
 		return nil, err
 	}
-	ctx, cancel := context.WithTimeout(context.Background(), s.config.ConnectTimeout)
-	defer cancel()
-	// slightly gross polling hack so that we shutdown quickly
-	go func() {
```

Comment:
Why can we get rid of this now? I'd expect this to be replaced with something like

```go
go func() {
  select {
    case <-ctx.Done():
    case <-s.shutdownCtx.Done():
      cancel()
  }
}

To make sure we interrupt both on `ctx` getting closed as well as `shutdownCtx` being cancelled.

Reply:
I think it's only fine to drop if this method is not exposed. If someone can call it who doesn't have access to `shutdownCtx` we need to explicitly abort.